Source/loadsave.cpp

```diff
@@ -183,6 +183,23 @@ class SaveHelper {
 
 }
 
+void RemoveInvalidItem(ItemStruct *pItem)
+{
+	bool isInvalid = !IsItemAvailable(pItem->IDidx) || !IsUniqueAvailable(pItem->_iUid);
+
+	if (!gbIsHellfire) {
+		isInvalid = isInvalid || (pItem->_itype == ITYPE_STAFF && GetSpellStaffLevel(pItem->_iSpell) == -1);
+		isInvalid = isInvalid || (pItem->_iMiscId == IMISC_BOOK && GetSpellBookLevel(pItem->_iSpell) == -1);
+		isInvalid = isInvalid || pItem->_iDamAcFlags != 0;
+		isInvalid = isInvalid || pItem->_iPrePower > IPL_AC_CURSE;
+		isInvalid = isInvalid || pItem->_iSufPower > IPL_AC_CURSE;

```

Comment:
I'm guessing IPL_AC_CURSE is the last valid affix? That's where I was thinking about some stuff other than hardcoded one